Scripts/Source/minai_Sex.psc

```diff
@@ -1359,3 +1307,79 @@ int function GetGender(actor targetActor)
 
   return sex
 endfunction
+
+function onSexStart(int ThreadID, string framework)
+  bool playerInvolved = isPlayerInvolved(ThreadID, framework)
+
+  if playerInvolved
+    AIFF.ChillOut()
+  EndIf
+
+  ; enable sex-mode for php
+  SetSexSceneState("on")
+  UpdateThreadTable("startthread", framework, ThreadID)
+
+  ; get actors to store random actor who will talk at the end. Doing it at start ensure that on sex end events we won't end up where in threads there are no actors anymore
+  Actor[] actors
+  if(framework == ostimType)
+    actors = OThread.GetActors(ThreadID)
+  elseif(framework == sexlabType)
+    actors = slf.GetController(ThreadID).Positions
+  endif
+
+  JMap.setForm(actorToSayOnEndMap, ThreadID, getWeightedRandomActorToSpeak(actors))
```

Comment:
I implemented it for ostim firstly, when ostim thread ends and fires event there are no actors anymore, so you can't get actor for post scene comment. So I decided to store it per thread on scene start(anyway we get random actor).

Now this logic is applied to sexlab flow as well.

Reply:
Makes sense!